Create react-native.config.js

MaxToyberman opened the pull request:
For react native 0.60 (auto linking), wont work otherwise

douglasjunior commented:
Great, thanks!

zoontek commented:
Autolinking worked great in version ``. The `react-native.config.js` file was useless because the `podspec` is located in the root directory and the android `AndroidManifest.xml` doesn't exists.

Now that the change has been done, the autolinking is broken. Check the config with the new cli: `yarn react-native config`. A revert or a deprecation could be nice.

douglasjunior commented:
Thanks @zoontek , I will revert and take a look more closely on this.

douglasjunior commented:
@zoontek do you have the link to any documentation on this?

zoontek commented:
@douglasjunior Yes! https://github.com/react-native-community/cli/blob/master/docs/autolinking.md

```

zoontek commented:
<img width="1425" alt="Screenshot 2019-07-18 at 15 01 18" src="https://user-images.githubusercontent.com/1902323/61459401-eed4db00-a96c-11e9-9c7b-5dc3c8b1b211.png">

This is the config created by the CLI for `4.0.13-13`. Everything is already OK 👌🏼

douglasjunior commented:
Thanks for the information @zoontek.

MaxToyberman commented:
it doesn't work for me without the config, I have many duplicate symbols

![Screen Shot 2019-07-18 at 16 14 25](https://user-images.githubusercontent.com/11058371/61460308-2183d100-a977-11e9-89ae-ad899285d9ed.png)

zoontek commented:
Did you unlink before the update?

MaxToyberman commented:
@zoontek yes

zoontek commented:
I made a quick update of the project with a working react native 0.60 example, no config file needed. Check it 🙂

Maybe check your header search paths? Did you clean your build folder? (option + click on `Product`) It must be tied to your project in some way.

<img width="263" alt="Screenshot 2019-07-18 at 15 44 34" src="https://user-images.githubusercontent.com/1902323/61462543-13cc4c80-a973-11e9-9c1e-14c321c1da27.png">

MaxToyberman commented:
@zoontek  doesn't help, when i run pod install both ReactNativeKeyboardManager and IQKeyboardManager are getting installed is this ok ?

zoontek commented:
@MaxToyberman Yes, that's the goal

MaxToyberman commented:
@zoontek strange, theres nothing in my header search paths + cleaned the project + deleted derived data

zoontek commented:
@MaxToyberman Can you run `node_modules/@react-native-community/cli/build/bin.js config` and give the `react-native-keyboard-manager` autolinking config?

MaxToyberman commented:
```
    "react-native-keyboard-manager": {
      "root": "/Users/maximtoyberman/projects/AIMobile2017/node_modules/react-native-keyboard-manager",
      "name": "react-native-keyboard-manager",
      "platforms": {
        "ios": {
          "sourceDir": "/Users/maximtoyberman/projects/AIMobile2017/node_modules/react-native-keyboard-manager/ios",
          "folder": "/Users/maximtoyberman/projects/AIMobile2017/node_modules/react-native-keyboard-manager",
          "pbxprojPath": "/Users/maximtoyberman/projects/AIMobile2017/node_modules/react-native-keyboard-manager/ios/ReactNativeKeyboardManager.xcodeproj/project.pbxproj",
          "podfile": null,
          "podspecPath": "/Users/maximtoyberman/projects/AIMobile2017/node_modules/react-native-keyboard-manager/ReactNativeKeyboardManager.podspec",
          "projectPath": "/Users/maximtoyberman/projects/AIMobile2017/node_modules/react-native-keyboard-manager/ios/ReactNativeKeyboardManager.xcodeproj",
          "projectName": "ReactNativeKeyboardManager.xcodeproj",
          "libraryFolder": "Libraries",
          "sharedLibraries": [],
          "plist": []
        },
        "android": null
      },
      "assets": [],
      "hooks": {},
      "params": []
    }
```

MaxToyberman commented:
@zoontek that's it , everything is working thank you.

douglasjunior commented:
@MaxToyberman What exactly that solved the problem? If this becomes a common problem I can add in README.

MaxToyberman commented:
@douglasjunior unlinked again, removed it from Header Search Paths and Library Search paths , both from the project and from the target, deleted Pods and pod install again.
